### PR TITLE
Support Vue JSX

### LINKED ISSUE #4450: Bug: not supporting vue tsx
### What version of `astro` are you using?

^1.0.8

### Are you using an SSR adapter? If so, which one?

none

### What package manager are you using?

pnpm

### What operating system are you using?

mac

### Describe the Bug

error： Cannot read properties of undefined (reading '0')

please check the reproducible example

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-v6q6m9?file=src%2Fpages%2Findex.astro,src%2Fcomponents%2FCounter.tsx&on=stackblitz

### Participation

- [ ] I am willing to submit a pull request for this issue.

### PR DESCRIPTION
**Pull request by bluwy**
## Changes

Close #4450

Introduce `jsx: boolean` option for `@astrojs/vue`. If true, it would add two renderers for .vue and .jsx files.

I'm not sure if there's a way to combine that like preact compat does.

## Testing

added a vue jsx test

## Docs

Documented in the readme.

### CODE REVIEW

**Review by matthewp — APPROVED** [review]

**Comment by bluwy** [comment]
Fixed the intellisense issue. Good to go once I get docs approval!

**Review by matthewp — APPROVED** [review]

File: packages/integrations/vue/README.md
Comment (Jutanium) [inline packages/integrations/vue/README.md:103]:
```suggestion
__`astro.config.mjs`__

```js
```

Comment (Jutanium) [reply packages/integrations/vue/README.md:103]: Just so it shows up as the code tab name when we pull the readme into docs!

File: packages/integrations/vue/README.md
Comment (Jutanium) [inline packages/integrations/vue/README.md:114]:
```suggestion
This will enable rendering for both Vue and Vue JSX components. To customize the Vue JSX compiler, pass an options object instead of a boolean. See the `@vitejs/plugin-vue-jsx` [docs](https://github.com/vitejs/vite/tree/main/packages/plugin-vue-jsx) for more details.
```

**Comment by Jutanium** [comment]
I think it would be good to show an example of passing options instead of a boolean, as the linked docs are slightly different (there the options are passed directly to the plugin, whereas here they're passed to the `jsx: ` property, if I'm understanding correctly). Other than that and my couple comments, LGTM!

**Comment by bluwy** [comment]
Added an example to show passing options! 766ade7

**Pull request merged**
